lib/src/client.cpp

```diff
@@ -41,23 +41,22 @@ namespace konstructs {
         return error_message;
     }
 
-    void Client::open_connection(const string &nick, const string &hash,
-                                 const string &hostname, const int port) {
+    void Client::open_connection(Settings settings) {
```

Comment:
Maybe it could take "Settings::Server" instead so we can remove "settings.server" in the code?